src/hooks/useCalculate.tsx

```diff
@@ -0,0 +1,82 @@
+import React, {useState} from "react";
+import {REG_RULES} from "../constants/RegExps";
+import {OperatorionsType} from "../models/Operations";
+import {getTotal} from "../utils/calculate";
+
+type useCalculateResultTuple = [
+  [number | null, number],
+  (value: OperatorionsType | number) => void,
+  () => void
+];
```

Comment:
반환되는 타입에 대한 적절한 명칭없이 해당 타입만 반환되고 있어 각각 어떤 역할을 가진 타입인지 추론이 쉽지 않습니다. 되도록 key, value 쌍으로 된 인터페이스로 작성하시는 것이 좋을 듯하고 현재의 구현이라면 이름을 붙여주는 것이 괜찮을 듯 합니다.

```suggestion
type reducer = [number | null, number]
type onPressValue = (value: OperatorionsType | number) => void
type reset = () => void
type useCalculateResultTuple = [
  reducer,
  onPressValue,
  reset
];
```